recipe/ingredients.py

```diff
@@ -77,10 +84,12 @@ def __init__(self, **kwargs):
         self.column_suffixes = kwargs.pop("column_suffixes", None)
         self.cache_context = kwargs.pop("cache_context", "")
         self.anonymize = False
+        self.roles = {}
         self._labels = []
 
         # What order should this be in
         self.ordering = kwargs.pop("ordering", "asc")
+        self.group_by_strategy = kwargs.pop("group_by_strategy", "labels")
```

Comment:
Group by strategy and roles were moved from the Dimension subclass to the Ingredient base class.